docs/src/paradox/03-apis/api-v2/query-language.md

```diff
@@ -182,81 +214,179 @@ is already known, e.g. to request specific information about that resource and
 perhaps about linked resources. In this case, the IRI of the main resource must
 be assigned to a variable using `BIND`.
 
-#### Properties
-
 Properties can be represented by an IRI or a query variable. If a
 property is represented by a query variable, it can be restricted to
 certain property IRIs using a `FILTER`.
 
-#### Values
+A Knora value (i.e. a value attached to a `knora-api:Resource`)
+must be represented as a query variable.
+
+### Filtering on Values
+
+#### Filtering on Values in the Simple Schema
+
+In the simple schema, a variable representing a Knora value can be used
+directly in a `FILTER` expression. For example:
+
+```
+?book incunabula:title ?title .
+FILTER(?title = "Zeitglöcklein des Lebens und Leidens Christi")
+```
+
+Here the type of `?title` is `xsd:string`.
+
+The following Knora value types can be compared with literals in `FILTER`
+expressions in the simple schema:
 
-Values can only be represented by a query variable. Value literals are
-currently not supported as the objects of statement patterns in the
-query. To restrict a value, a FILTER must be used. Without a FILTER, all
-the instances of a value are returned.
+- Text values (`xsd:string`)
+- Integer values (`xsd:integer`)
+- Decimal values (`xsd:decimal`)
+- Boolean values (`xsd:boolean`)
+- Date values (`knora-api:Date`)
 
-#### Functions in FILTER Expressions
+A Knora value may not be represented as the literal object of a predicate;
```

Comment:
I think we should see how hard it is to allow literals in the simple schema. The main reason why we forbade this was because it makes the processing harder in `SearchResponderV2`.

In simple schema, we have to add an extra level by generating a statement for `valueHasString`. However, from the user's perspective it is rather hard to  understand why Gravsearch supports literals in complex schema while it does not in simple schema.

Reply:
If you used a literal instead of a variable, that would mean you couldn't mention that value in the CONSTRUCT clause, so wouldn't it be excluded from the results? I think this is rarely what the user would want.

Reply:
By the way, the rule is currently the same for both schemas: you can't represent a Knora value as the literal object of a predicate. In the complex schema, you can use literals for things below the value level, but not for the value itself.

Reply:
> If you used a literal instead of a variable, that would mean you couldn't mention that value in the CONSTRUCT clause, so wouldn't it be excluded from the results?

I think in that case you would just use the same statement:

```sparql
PREFIX beol: <http://0.0.0.0:3333/ontology/0801/beol/simple/v2#>
PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

    CONSTRUCT {
        ?letter knora-api:isMainResource true .

        ?letter beol:creationDate ?date .

        ?letter ?linkingProp1  ?person1 .

        ?person1 beol:hasFamilyName "Meier" .

    } WHERE {
      
        ?letter a beol:letter .

        ?letter beol:creationDate ?date .

        ?letter ?linkingProp1 ?person1 .
        FILTER(?linkingProp1 = beol:hasAuthor || ?linkingProp1 = beol:hasRecipient )

        ?person1 beol:hasFamilyName "Meier".

    } ORDER BY ?date
```

I think in a first version of Gravsearch, we allowed for literals. But when I refactored it, I did not support them anymore because the processing logic would have been more complex.

In `webapi/src/test/scala/org/knora/webapi/e2e/v2/SearchRouteV2R2RSpec.scala` there are still some old tests involving literals that are ignored.

Reply:
OK, let's look at this after standoff search is done.